apps/user-office-frontend/src/components/questionary/questionaryComponents/FileUpload/QuestionaryComponentFileUpload.tsx

```diff
@@ -1,70 +1,71 @@
-  props: BasicComponentProps & { files?: string[] }
-) {
-  const {
-    answer,
-    onComplete,
-    formikProps: { errors },
-  } = props;
-  const {
-    question: { id },
-    value,
-  } = answer;
-  const isError = errors[id] ? true : false;
-  const config = answer.config as FileUploadConfig;
-  const [stateValue, setStateValue] =
-    useState<FileIdWithCaptionAndFigure[]>(value);
-
-  useEffect(() => {
-    setStateValue(answer.value);
-  }, [answer]);
-
-  return (
-    <FormControl
-      error={isError}
-      required={config.required}
-      margin="dense"
-      fullWidth
-    >
-      <FormLabel>
-        {answer.question.question}
-        {config.small_label && (
-          <>
-            <br />
-            <small>{config.small_label}</small>
-          </>
-        )}
-      </FormLabel>
-      <FileUploadComponent
-        id={answer.question.id}
-        maxFiles={config.max_files}
-        pdfPageLimit={config.pdf_page_limit}
-        fileType={config.file_type ? config.file_type.join(',') : ''}
-        onChange={(fileMetaDataList: FileIdWithCaptionAndFigure[]) => {
-          const newStateValue = fileMetaDataList.map((file) => ({
-            id: file.id,
-            caption: file.caption,
-            figure: file.figure,
-          }));
-
-          setStateValue(newStateValue);
-          onComplete(newStateValue);
-        }}
-        value={stateValue}
-      />
-      {isError && <FormHelperText>{errors[id]}</FormHelperText>}
-    </FormControl>
-  );
-}
+import FormControl from '@mui/material/FormControl';
+import FormHelperText from '@mui/material/FormHelperText';
+import FormLabel from '@mui/material/FormLabel';
+import React, { useEffect, useState } from 'react';
+
+import {
+  FileIdWithCaptionAndFigure,
+  FileUploadComponent,
+} from 'components/common/FileUploadComponent';
+import { BasicComponentProps } from 'components/proposal/IBasicComponentProps';
+import { FileUploadConfig } from 'generated/sdk';
+
+export function QuestionaryComponentFileUpload(
+  props: BasicComponentProps & { files?: string[] }
+) {
+  const {
+    answer,
+    onComplete,
+    formikProps: { errors },
+  } = props;
+  const {
+    question: { id },
+    value,
+  } = answer;
+  const isError = errors[id] ? true : false;
+  const config = answer.config as FileUploadConfig;
+  const [stateValue, setStateValue] =
+    useState<FileIdWithCaptionAndFigure[]>(value);
+
+  useEffect(() => {
+    setStateValue(answer.value);
+  }, [answer]);
+
+  return (
+    <FormControl
+      error={isError}
+      required={config.required}
+      margin="dense"
+      fullWidth
+    >
+      <FormLabel>
+        {answer.question.question}
+        {config.small_label && (
+          <>
+            <br />
+            <small>{config.small_label}</small>
+          </>
+        )}
+      </FormLabel>
+      <FileUploadComponent
+        id={answer.question.id}
+        question={answer.question.question}

```

Comment:
I think it is better to use answer.question.id here, because id is a unique identifier

Reply:
I am reverting these changes and will just select the element by question name itself. Something like:
`cy.contains(fileQuestion).parent().find('input[type="file"]')`. Thanks for the good comment.